docs/root/operations/admin.rst

```diff
@@ -647,7 +667,55 @@ modify different aspects of the server:
       ]
     }
 
-.. http:get:: /stats?format=prometheus
+  .. http:get:: /stats?format=json&histogram_buckets=detailed
+
+  Shows histograms as both percentile summary data, and raw bucket
+  data, up to 64 buckets. If the raw bucket data has more than 64
+  buckets, then adjacent buckets are combined until 64 is reached.
+
+  Example output:
+
+  .. code-block:: json
+
+    {
+      "stats": [
+        {
+          "histograms": {
+            "supported_percentiles": [0, 25, 50, 75, 90, 95, 99, 99.5, 99.9, 100],
+            "details": [
+              {
+                "name": "http.admin.downstream_rq_time",
+                "percentiles": [
+                  { "interval": null, "cumulative": 1 },
+                  { "interval": null, "cumulative": 1.0351851851851852 },
+                  { "interval": null "cumulative": 1.0703703703703704 },
+                  { "interval": null, "cumulative": 2.0136363636363637 },
+                  { "interval": null "cumulative": 2.0654545454545454 },
+                  { "interval": null "cumulative": 2.0827272727272725 },
+                  { "interval": null "cumulative": 2.0965454545454545 },
+                  { "interval": null, "cumulative": 2.098272727272727 },
+                  { "interval": null, "cumulative": 2.0996545454545457 },
+                  { "interval": null "cumulative": 2.1 }
+                ],
+                "totals": [
+                  { "min_value": 1, "width": 0.25, "count": 25 },
```

Comment:
I think width is technically redundant (it should refer to the next bucket's lower bound). But if it makes processing easier, I'm fine leaving it in.

Is this example real output or contrived? If it's real, I'm very surprised. If it's contrived, please make the buckets contiguous (no gaps between buckets) so the example isn't misleading.

Reply:
I think this data was not contrived; it's real.

You can also see this in stats_render_test.cc:335
```
                "totals": [
                    {"min_value": 200, "width": 10, "count": 1},
                    {"min_value": 300, "width": 10, "count": 2}],
                "intervals":[
                    {"min_value": 200, "width": 10, "count": 1},
                    {"min_value": 300, "width": 10, "count": 2}]}]}}]}
```
that's real data from circlhist and the input values `{200, 300, 300}` in the test there.

bucket2.min_value is not the same as bucket1.min_value + bucket1.width. There is space between the buckets.

Reply:
Interesting! Then definitely leave as-is.